File: VoxAnvil/src/main/java/com/qualityplus/anvil/base/config/impl/AnvilFiles.java
package com.qualityplus.anvil.base.config.impl;

import com.qualityplus.anvil.api.config.ConfigFiles;
import com.qualityplus.anvil.base.config.*;
import com.qualityplus.assistant.lib.eu.okaeri.injector.annotation.Inject;
import com.qualityplus.assistant.lib.eu.okaeri.platform.core.annotation.Component;

@Component
public final class AnvilFiles implements ConfigFiles<Config, Inventories, Messages, Commands> {
    private @Inject Inventories inventories;
    private @Inject Messages messages;
    private @Inject Commands commands;
    private @Inject Config config;

    @Override
    public Config config() {
        return config;
    }

    @Override
    public Inventories inventories() {
        return inventories;
    }

    @Override
    public Messages messages() {
        return messages;
    }

    @Override
    public Commands commands() {
        return commands;
    }

    @Override
    public void reloadFiles() {
        config.load();
        messages.load();
        inventories.load();
        commands.load();
    }
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/base/config/impl/AnvilFiles.kt
package com.qualityplus.anvil.base.config.impl

import com.qualityplus.anvil.api.config.ConfigFiles
import com.qualityplus.anvil.base.config.Commands
import com.qualityplus.anvil.base.config.Config
import com.qualityplus.anvil.base.config.Inventories
import com.qualityplus.anvil.base.config.Messages
import eu.okaeri.injector.annotation.Inject
import eu.okaeri.platform.core.annotation.Component

@Component
class AnvilFiles : ConfigFiles<Config?, Inventories?, Messages?, Commands?> {
    @Inject
    private val inventories: Inventories? = null

    @Inject
    private val messages: Messages? = null

    @Inject
    private val commands: Commands? = null

    @Inject
    private val config: Config? = null
    override fun config(): Config? {
        return config
    }

    override fun inventories(): Inventories? {
        return inventories
    }

    override fun messages(): Messages? {
        return messages
    }

    override fun commands(): Commands? {
        return commands
    }

    override fun reloadFiles() {
        config!!.load()
        messages!!.load()
        inventories!!.load()
        commands!!.load()
    }
}